src/test/java/io/github/christophermanahan/carnitas/HTTPServerTest.java

```diff
@@ -100,44 +97,53 @@ public void log(String message) {
         }
     }
 
-    private class TestListener implements Listener {
-
-        private final Connection connection;
+    private class TestAcceptor implements Acceptor {
+        private final TestConnection connection;
+        int receiveCount;
 
-        public TestListener(Connection connection) {
+        TestAcceptor(TestConnection connection, int receiveCount) {
             this.connection = connection;
+            this.receiveCount = receiveCount;
         }
 
-        public Connection listen() {
+        public TestConnection accept() {
+            connection.open();
+            receiveCount -= 1;
```

Comment:
I have some ideas on how to make this a little more elegant. Maybe we can pair on that this Friday?

Reply:
But not a blocker for this PR.

Reply:
Oh, well, we should still pair on it, but I switched it over to an iterator of connections after seeing the previous comment!